modules/ROOT/pages/provisioning-oraclecloud.adoc

```diff
@@ -0,0 +1,326 @@
+. Upload the image to a bucket:
++
+[source, bash]
+----
+oci os object put --bucket-name "${bucket_name}" --file "${image_name}"
+----
++
+. Import the image as a custom image and remember its ID:
++
+[source, bash]
+----
+namespace="$(oci os ns get | jq -r '.data')"
+image_id="$(oci compute image import from-object \
+    --compartment-id "${compartment_ocid}" \
+    --namespace "${namespace}" \
+    --bucket-name "${bucket_name}" \
+    --name "${image_name}" \
+    --display-name "Fedora CoreOS" \
+    --launch-mode PARAVIRTUALIZED \
+    --source-image-type QCOW2 \
+    --operating-system "Linux" \
+    | jq -r '.data.id')"
+----
++
+. Wait until the import is completed. To list all imported FCOS images:
++
+[source, bash]
+----
+oci compute image list --compartment-id "${compartment_ocid}" --display-name "Fedora CoreOS"
+----
++
+. Configure image capabilities
++
+[source, bash]
+----
+global_cap_id="$(oci compute global-image-capability-schema list | jq -r '.data[0].id')"
+global_cap_version_name="$(oci compute global-image-capability-schema-version list \
+    --global-image-capability-schema-id ${global_cap_id} \
+    --display-name 1.2 \
+    | jq -r '.data[0].name')"
+
+oci compute image-capability-schema create \
+    --compartment-id "${compartment_ocid}" \
+    --image-id "${image_id}" \
+    --global-image-capability-schema-version-name "${global_cap_version_name}" \
+    --schema-data '{
+    "Compute.AMD_SecureEncryptedVirtualization": {
+        "default-value": true,
+        "descriptor-type": "boolean",
+        "source": "IMAGE"
+    },
+    "Compute.Firmware": {
+        "default-value": "UEFI_64",
+        "descriptor-type": "enumstring",
+        "source": "IMAGE",
+        "values": [
+            "BIOS",
+            "UEFI_64"
+        ]
+    },
+    "Compute.SecureBoot": {
+        "default-value": true,
+        "descriptor-type": "boolean",
+        "source": "IMAGE"
+    },
+    "Storage.Iscsi.MultipathDeviceSupported": {
+        "default-value": true,
+        "descriptor-type": "boolean",
+        "source": "IMAGE"
+    }
+}'
+----
++
+. Mark the image as compatible with all https://docs.oracle.com/en-us/iaas/Content/Compute/References/computeshapes.htm[shapes].
++
+.Remove default compatible shapes
+[source, bash]
+----
+for shape in $(oci compute image-shape-compatibility-entry list --image-id "${image_id}" | jq -r '.data[].shape' | grep -v '.Generic'); do
+    echo "Removing ${shape}"
+    oci compute image-shape-compatibility-entry remove --force --image-id "${image_id}" --shape-name "${shape}"
+done
+----
++
+.Mark as compatible with x86_64 VM shapes
+[source, bash]
+----
+shapes_amd64=(
+    "VM.Standard3.Flex"
+    "VM.Standard.E2.1.Micro"
+    "VM.Standard.E4.Flex"
+    "VM.Standard.E5.Flex"
+    "VM.Standard.E6.Flex"
+    "VM.DenseIO.E4.Flex"
+    "VM.DenseIO.E5.Flex"
+    "VM.GPU2.1"
+    "VM.GPU3.1"
+    "VM.GPU3.2"
+    "VM.GPU3.4"
+    "VM.GPU.A10.1"
+    "VM.GPU.A10.2"
+    "VM.Optimized3.Flex"
+)
```

Comment:
These lists seem prone to get out of date. Is there any way to dynamically determine what values to use here?

I'm sure you've investigated, but there isn't some way to mark it as compatible with all shapes without having to provide all the strings for each one?